### PR TITLE
disk_partitioner filesystem creation implementation

### PR DESCRIPTION
**Pull request by xek**

### CODE REVIEW

File: ironic-lib/0004-Filesystem-creation-implementation.patch [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
++            fstype = disk.get('fstype', 'ext4')
++            device = devices.get(disk_name, disk_name)
++            if not disk.get('preserve', False):
++                if fstype == 'swap':
++                    out, err = utils.execute('mkswap', device)
++                else:
++                    out, err = utils.execute('yes | mkfs.{} {}'.format(
++                        fstype, device), shell=True)
++                LOG.debug("mkfs stdout: {}".format(out))
++                LOG.debug("mkfs stderr: {}".format(err))
```
Comment (sagarun) [inline ironic-lib/0004-Filesystem-creation-implementation.patch]: if there is an error in mkfs, we should stop the installation right?

**Review by sagarun — CHANGES_REQUESTED** [review]
Please address the review comments

File: ironic-lib/0004-Filesystem-creation-implementation.patch (hunk 1 above)
Comment (xek) [reply ironic-lib/0004-Filesystem-creation-implementation.patch]: Yes, an exception is raised by utils.execute in this case.

**Review by sagarun — APPROVED** [review]

**Comment by jimrollenhagen** [comment]
Merging this, not fully tested (waiting for hardware) but will clean up later as needed.

**Pull request merged**